### PR TITLE
Add input validation for the sign of rate parameters

### PR DESCRIPTION
**Pull request by ovidharshini**
Adds checks to test if Duration is positive and Money is non-negative.

### CODE REVIEW

File: src/main/java/peoplesoft/model/job/Money.java [hunk 1]
```diff
@@ -37,6 +39,15 @@ public Money(BigDecimal value) {
         this.value = value.setScale(VALUE_SCALE, RoundingMode.HALF_UP);
     }
 
+    /**
+     * Throws NegativeMoneyException if {@code money} is negative.
+     */
+    public static void requireNonNegative(Money money) {
```
Comment (zhongfu) [inline src/main/java/peoplesoft/model/job/Money.java:45]: would it be better to have it in a separate util class, eg `MoneyUtil`? i think some of the existing model classes do this

File: src/main/java/peoplesoft/model/job/Money.java [hunk 2]
```diff
@@ -37,6 +39,15 @@ public Money(BigDecimal value) {
...
+        if (money.value.compareTo(new BigDecimal("0")) < 0) {
```
Comment (zhongfu) [inline src/main/java/peoplesoft/model/job/Money.java:46]: try [`signum()`](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#signum--)

File: src/main/java/peoplesoft/model/job/Rate.java [hunk 3]
```diff
@@ -17,14 +20,26 @@
...
+        requirePositive(duration);
         this.amount = amount;
         this.duration = duration;
     }
 
+    /**
+     * Throws NonPositiveDurationException if {@code duration} is not positive.
+     */
+    private void requirePositive(Duration duration) {
+        if (duration.compareTo(Duration.ofHours(0)) <= 0) {
```
Comment (zhongfu) [inline src/main/java/peoplesoft/model/job/Rate.java:38]: [`isNegative()`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#isNegative--)

File: src/main/java/peoplesoft/model/job/Rate.java [hunk 4]
```diff
@@ -69,7 +84,7 @@ public int hashCode() {
     /**
      * Prints the 2 decimal place currency format of the value.
      *
-     * @returns Value in currency format as a string.
+     * @return Value in currency format as a string.
```
Comment (zhongfu) [inline src/main/java/peoplesoft/model/job/Rate.java:87]: good catch :+1:

File: src/main/java/peoplesoft/model/job/exceptions/NegativeMoneyException.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package peoplesoft.model.job.exceptions;
+
+/**
+ * Signals that the operation will result in negative rate.
+ */
+public class NegativeMoneyException extends RuntimeException {
```
Comment (zhongfu) [inline src/main/java/peoplesoft/model/job/exceptions/NegativeMoneyException.java:6]: `NegativeMoneyValueException`? would that be better?

File: src/main/java/peoplesoft/model/job/Rate.java (hunk 3 above)
Comment (zhongfu) [reply src/main/java/peoplesoft/model/job/Rate.java:38]: oops, you might want to either do `isNegative() || isZero()` or compare against `Duration.ZERO`

**Review by zhongfu — CHANGES_REQUESTED** [review]

**Review by zhongfu — COMMENTED** [review]
consider adding tests too, so that we'll catch it early if we accidentally introduce regressions

File: src/main/java/peoplesoft/model/job/Money.java (hunk 1 above)
Comment (ovidharshini) [reply src/main/java/peoplesoft/model/job/Money.java]: I have added a equivalent class for Duration as well. For now, the Util classes can be found under model > job. I'm not sure if there should be a common Util folder for all models instead.

**Review by zhongfu — APPROVED** [review]
looks good

**Pull request merged**